analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -79,86 +62,85 @@ protected sealed override SymbolReferenceInfo CreateMessage(SyntaxTree syntaxTre
             base.ShouldGenerateMetrics(tree)
             && !HasTooManyTokens(tree);
 
-        protected virtual SyntaxToken? GetSetKeyword(ISymbol valuePropertySymbol) => null;
-
-        protected static bool IsValuePropertyParameter(ISymbol symbol) =>
-            symbol is IParameterSymbol {IsImplicitlyDeclared: true, Name: "value"};
-
-        private SymbolReferenceInfo.Types.SymbolReference GetSymbolReference(SymRefInfo[] allReference, SyntaxTree tree)
+        private IEnumerable<ReferenceInfo> GetReferences(SyntaxNode root, SemanticModel model)
         {
-            TextSpan declarationSpan;
-            if (allReference.FirstOrDefault(r => r.IsDeclaration) is { } declaration)
-            {
-                declarationSpan = declaration.IdentifierToken.Span;
-            }
-            else
+            var references = new List<ReferenceInfo>();
+            var knownIdentifiers = new HashSet<string>();
```

Comment:
What about case insensitivity for VB.NET?
```suggestion
            var knownIdentifiers = new HashSet<string>(NameComparison.NameComparison);
```

Reply:
..an UT is probably missing

Reply:
No, I'm working on it. I've pushed since I did a rebase and wanted to see the coverage. I did not ask for review yet 😃

Reply:
We are doing it even without asking  😄